### ISSUE TITLE
bug in local_efficency 

### ISSUE THREAD

**Issue by idc9** [networkx/networkx#2233]
The local_efficiency function is not computed correctly (I'm using python 2, networkx version 1.11)

The bug is in line 140 of [efficiency.py](https://github.com/networkx/networkx/blob/master/networkx/algorithms/efficiency.py) which is currently
`return sum(global_efficiency(G.subgraph(G[v])) for v in G) / len(G)`

The issue is `G.subgraph(G[v])` returns the neighborhood subgraph of v **excluding** v. This is not the correct computation and, for example, leads to an error

```
G = nx.Graph()  # triangle plus one node

G.add_nodes_from([1, 2, 3, 4])

G.add_edge(1, 2)
G.add_edge(1, 3)
G.add_edge(2, 3)
G.add_edge(1, 4)

local_efficiency(G)
```

raises
`NetworkXNoPath: No path between 2 and 4.`
When local_efficency calls global_efficiency on `G.subgraph(G[1])` it gets a disconnected graph. The fix for this error is simple i.e. replace line 140 with

`return sum(global_efficiency(nx.ego_graph(G, v)) for v in G) / len(G)`

(On a side note the efficiency.py file did not come with my networkx package -- I downloaded it directly from Github -- not sure why this happened?)

**Comment by dschult** [comment]
Efficiency was added after v1.11, so that's why the module was not included in your network package.

**Comment by jfinkels** [comment]
Fixed in pull request #2246. Thanks!
